### PR TITLE
Fall back on `sys.real_prefix` for Python 2.7

### PR DESCRIPTION
**Pull request by dandersson**
`sys.base_prefix` is used in the SQLite3 import hook from pull request #261. This attribute was introduced in Python 3.3, which makes this hook raise an uncaught exception when running Python 2.7.

Fall back to looking at `sys.real_prefix`, which is set by `virtualenv` to the equivalent of `sys.base_prefix` when running in a virtual environment, or just `sys.prefix` otherwise.

Since this regards Python 2.7 compatibility, this should only be relevant for the cx_Freeze 5.x branch.

Sample traceback from the kind of error this intends to fix:

```
running build
running build_exe
Traceback (most recent call last):
  File "config/cx_freeze_setup.py", line 315, in <module>
    main()
  File "config/cx_freeze_setup.py", line 287, in main
    executables=executables
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\dist.py", line 349, in setup
    distutils.core.setup(**attrs)
  File "C:\python27\Lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "C:\python27\Lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "C:\python27\Lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\python27\Lib\distutils\command\build.py", line 127, in run
    self.run_command(cmd_name)
  File "C:\python27\Lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\python27\Lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\dist.py", line 219, in run
    freezer.Freeze()
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\freezer.py", line 621, in Freeze
    self.finder = self._GetModuleFinder()
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\freezer.py", line 340, in _GetModuleFinder
    finder.IncludePackage(name)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 655, in IncludePackage
    self._ImportAllSubModules(module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 287, in _ImportAllSubModules
    recursive)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 279, in _ImportAllSubModules
    deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 569, in _ScanCode
    fromList, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 210, in _EnsureFromList
    self._ImportModule(subModuleName, deferredImports, caller)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 416, in _LoadModule
    namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 485, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 384, in _InternalImportModule
    namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 416, in _LoadModule
    namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 485, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 416, in _LoadModule
    namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 485, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 345, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 326, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 416, in _LoadModule
    namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 485, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 569, in _ScanCode
    fromList, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 210, in _EnsureFromList
    self._ImportModule(subModuleName, deferredImports, caller)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 474, in _LoadModule
    self._ScanCode(module.code, module, deferredImports)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 588, in _ScanCode
    topLevel = False)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 564, in _ScanCode
    module, relativeImportIndex)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 310, in _ImportModule
    deferredImports, namespace = namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 403, in _InternalImportModule
    parentModule, namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 416, in _LoadModule
    namespace)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 485, in _LoadPackage
    self._LoadModule(name, fp, path, info, deferredImports, parent)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 463, in _LoadModule
    self._RunHook("load", module.name, module)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\finder.py", line 536, in _RunHook
    method(self, *args)
  File "C:\dev\test_sqlite3_hook\env\lib\site-packages\cx_Freeze\hooks.py", line 826, in load_sqlite3
    dll_path = os.path.join(sys.base_prefix, "DLLs", dll_name)
AttributeError: 'module' object has no attribute 'base_prefix'
```

### CODE REVIEW

**Comment by FreeHarry** [comment]
@dandersson: Thanks for improving my pull request #261! I forgot to check it for older python versions.

File: cx_Freeze/hooks.py [hunk 1]
```diff
@@ -816,12 +816,23 @@ def load_clr(finder, module):
...
+
+def _get_base_prefix():
+    """Get prefix to site-specific directory prefix where the platform
+    independent Python files are installed."""
+    try:
+        return sys.base_prefix  # Python 3.3+
+    except AttributeError:
+        # For Python 2, use sys.real_prefix set by virtualenv if set,
+        # otherwise fall back to sys.prefix
+        return getattr(sys, "real_prefix", "prefix")
```
Comment (anthony-tuininga) [inline cx_Freeze/hooks.py]: This code doesn't do what I think you want it to do -- namely, use sys.real_prefix unless it cannot be found, in which case return sys.prefix. Perhaps something like this instead?

return getattr(sys, "real_prefix", sys.prefix)

Comment (dandersson) [reply cx_Freeze/hooks.py]: Absolutely correct -- I will update the pull request.

**Comment by wlach** [comment]
@anthony-tuininga would it be possible to release a new version of the library in the 5.x series with this fix? I'd like to use a later version with mozregression, which is stuck on 2.7 for now.

**Comment by anthony-tuininga** [comment]
Yes. Will release a new version of cx_Freeze 5.x soon. Its overdue. :-)

**Comment by lcunha0118** [comment]
@anthony-tuininga I have version 5.0 and I still have the same error. And the hooks.py file was modified so the same fix does not apply. Was the problem fixed?

**Comment by tnsolley** [comment]
I can confirm this issue is still present in version 5.02 of cx_Freeze.

**Comment by mattbk** [comment]
> I can confirm this issue is still present in version 5.02 of cx_Freeze.

Ditto.

**Comment by ourish** [comment]
> I can confirm this issue is still present in version 5.02 of cx_Freeze.

As of 3/11 the issue remains

**Pull request merged**